modules/core/src/main/java/org/apache/ignite/internal/visor/cache/index/IndexForceRebuildTask.java

```diff
@@ -68,111 +67,99 @@ protected IndexForceRebuildJob(@Nullable IndexForceRebuildTaskArg arg, boolean d
             throws IgniteException
         {
             //Either cacheNames or cacheGrps must be specified
-            assert (arg.cacheNames() == null) != (arg.cacheGrps() == null) :
+            assert (arg.cacheNames() == null) ^ (arg.cacheGrps() == null) :
                 "Either cacheNames or cacheGroups must be specified";
 
```

Comment:
The code below is difficult to understand and needs to be simplified.

What needs to be done here:
- Collect contexts for which you want to rebuild indexes;
- Determine for which caches / groups contexts were not found;
- Send contexts for rebuilding via **IgniteCacheDatabaseSharedManager#forceRebuildIndexes**, its return value will allow you to understand for which contexts the rebuilding has not started since it is already in the process;
- Send a response to the utility.

By the name of the cache, you can get the contexts like this:
- ignite.cachex("cacheName").context();
- ignite.context().cache().cache("cacheName").context();

For the group id like this:
- ignite.context().cache().cacheGroup(0).caches()